pkg/runtime/wfengine/README.md

```diff
@@ -0,0 +1,175 @@
+# Dapr Workflow Engine
+
+The Dapr Workflow engine enables developers to author workflows using code and execute them using the Dapr sidecar. You can learn more about this project here: [[Proposal] Workflow building block and engine (#4576)](https://github.com/dapr/dapr/issues/4576).
+
+This README is designed to be used by maintainers to help with getting started. This README will be updated with more information as the project progresses.
+
+## Building Daprd
+
+The workflow engine is entirely encapsulated within the [dapr sidecar (a.k.a. daprd)](https://docs.dapr.io/concepts/dapr-services/sidecar/). All dependencies are compiled directly into the binary.
+
+Internally, this engine depends on the [Durable Task Framework for Go](https://github.com/microsoft/durabletask-go), an MIT-licensed open-source project for authoring workflows (or "orchestrations") as code. Use the following command to get the latest build of this dependency:
+
+```bash
+go get github.com/microsoft/durabletask-go
+```
+
+Be mindful, that the above command will also pull in dependencies for sqlite, which we don't want or require. Those can be manually removed from go.mod and go.sum.
+
+The following bash command can be used to build a version of Daprd that supports the workflow engine.
+
+```bash
+DEBUG=1 make build
+```
+* `DEBUG=1` is required to attach debuggers. This should never be set for production or performance testing workloads.
+
+After building, the following bash command can be run from the project root to test the code:
+
+```bash
+./dist/linux_amd64/debug/daprd --app-id foo --dapr-grpc-port 4001 --placement-host-address :50005 --components-path ~/.dapr/components/ --config ~/.dapr/config.yaml
```

Comment:
The "components-path" is completely optional here, correct? If so, do we want to keep it here? I ask because it may cause some confusion since all other engines, i.e. temporal need to have the component path specified whereas the built-in does not. It may lead to the users thinking that they need to have a yaml created for the built-in.

Reply:
Good points, but this is a slightly special case because we're running `daprd` directly from the enlistment root (something users won't normally do) rather than using the `dapr` CLI. I found that I did need to explicitly specify `--components-path`, not specifically because we needed it for a workflow component, but rather because we needed it for the _state store_ component (e.g. redis).

In the user-facing documentation, we won't explicitly include `--components-path`. We'll instead assume that they're just using the normal dapr initialization path, which includes a default redis state store setup. We can also mention explicitly that there is no component yaml required for the Dapr workflow engine.